conan/tools/cmake/cmakedeps.py

```diff
@@ -764,3 +669,81 @@ def _config(self, filename, name, version, public_deps_names):
                                           build_modules_block=build_modules_block,
                                           find_dependencies_block=find_dependencies_block)
         return tmp
+
+    def get_computed_names(self, requires):
+        """Convert the package and components names according to the cpp_info.names information
+        and cpp_info.filenames"""
+        _PackageNames = namedtuple("_PackageNames", ["name", "filename", "components"])
+
+        def _get_real_name(cpp_info, default_name):
+            _name = cpp_info.names.get(self.name)
+            if _name is not None:
+                return _name
+            find_name = cpp_info.names.get("cmake_find_package_multi")
+            if find_name is not None:
+                # Not displaying a warning, too noisy as this is called many times
+                conan_v2_error("'{}' defines information for 'cmake_find_package_multi', "
+                               "but not 'CMakeDeps'".format(default_name))
+                return find_name
+            return default_name
+
+        def _get_real_filename(cpp_info, default_name):
+            _filename = cpp_info.filenames.get(self.name)
+            if _filename is not None:
+                return _filename
+            find_name = cpp_info.filenames.get("cmake_find_package_multi")
+            if find_name is not None:
+                # Not displaying a warning, too noisy as this is called many times
+                conan_v2_error("'{}' defines information for 'cmake_find_package_multi', "
+                               "but not 'CMakeDeps'".format(default_name))
+                return find_name
+            return default_name
+
+        ret = {}
+        for req in requires:
+            name = _get_real_name(req.cpp_info, req.name)
+            filename = _get_real_filename(req.cpp_info, req.name)
+            components = {}
+            for comp_name, comp in req.cpp_info.components.items():
+                components[comp_name] = _get_real_name(comp, comp_name)
+            if req.name not in components:
+                # Default component (no component) => package name
+                components[req.name] = name
+            ret[req.name] = _PackageNames(name, filename, components)
+        return ret
+
+    @staticmethod
+    def get_target_names_and_filenames(req, cpp_info, computed_names):
```

Comment:
The management of names and filenames is being improved by @czoido in https://github.com/conan-io/conan/pull/8727, probably this will be simplified by that